### PR TITLE
Feature/mobile layout

### PR DESCRIPTION
**Pull request by pedrohw12**
Marketplace home page with responsive layout.

Screen Shot 2022-01-13 at 14 46 22

Marketplace/explorer

Screen Shot 2022-01-13 at 18 04 55

Marketplace/create

Screen Shot 2022-01-14 at 09 16 46

Marketplace/orders

Screen Shot 2022-01-14 at 12 01 02

Marketplace/orders/id

Screen Shot 2022-01-14 at 15 56 44

### CODE REVIEW

**Review by karacurt — COMMENTED** [review]
Nice!!

File: components/common/FilterMarketplace/Filters.tsx [hunk 1]
```diff
@@ -15,7 +20,12 @@ const Filters: React.FC = () => {
   ]
 
   return (
-    <FiltersContainer>
+    <FiltersContainer isVisible={isVisible}>
+      {isVisible && (
+        <button className='close-btn' onClick={onClose} type='button'>
+          X
+        </button>
```
Comment (juo-dev) [inline components/common/FilterMarketplace/Filters.tsx:27]: Maybe here create a component?

File: components/common/FilterMarketplace/styles.tsx [hunk 2]
```diff
@@ -1,49 +1,51 @@
...
-  @media only screen and (max-width: 600px) {
-    display: block;
-  }
-`
-
-export const FiltersContainer = styled.div`
+export const FiltersContainer = styled.div<FilterContainerProps>`
   height: 100%;
   width: 25%;
+  display: ${(props) => (props.isVisible ? 'flex' : 'none')};
```
Comment (juo-dev) [inline components/common/FilterMarketplace/styles.tsx:10]: My doubt: the best practices are to create css classes inside the components?

File: components/common/MHProductsList/index.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+  function handleNextProduct() {
+    if (currentProduct === orders.length - 1) return null
+    setCurrentProduct((oldState) => oldState + 1)
+    return null
+  }
+
+  function products() {
+    return (
+      <>
```
Comment (juo-dev) [inline components/common/MHProductsList/index.tsx:32]: What is it?

File: pages/marketplace/create.tsx [hunk 4]
```diff
@@ -35,32 +33,24 @@ const CreateOrder: NextPage = () => {
...
+            <Title>Create Order</Title>
+          </div>
+          <CreateOrderSection>
+            <Left>
+              <FormCreateOrder setOrder={setOrder} />
+            </Left>
+            <Right>
+              <CardCreateOrder order={order} />
+            </Right>
+          </CreateOrderSection>
```
Comment (juo-dev) [inline pages/marketplace/create.tsx:49]: I really like it when we remove code

File: pages/marketplace/index.tsx [hunk 5]
```diff
@@ -38,41 +35,35 @@ const MarketplaceHome: NextPage = () => {
...
+              and sell game assets such as Boosters, Mouse Heroes and Ghost Villains (coming soon)
+            </Subtitle>
+          </TitleContainer>
+          <ContainerButton>
+            <LinkButton className='right' href='/marketplace/explore'>
+              Explore now
+            </LinkButton>
+            <ButtonDark href='/marketplace/create'>Create order</ButtonDark>
+          </ContainerButton>
+        </CTASection>
```
Comment (juo-dev) [inline pages/marketplace/index.tsx:62]: Much better!

File: styles/marketplace/stylesCreate.ts [hunk 6]
```diff
@@ -107,54 +113,33 @@ export const Title = styled.div`
...
-    display: flex;
-  }
-
-  @media only screen and (min-device-width: 768px) and (max-device-width: 1024px) {
-    display: flex;
-  }
-
-  @media only screen and (max-width: 600px) {
-    display: none;
+  @media (min-width: 1000px) {
```
Comment (juo-dev) [inline styles/marketplace/stylesCreate.ts:141]: Great!

File: components/common/MHProductsList/index.tsx [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+import CardShop from '../CardShop'
+import { MHProducts, ButtonsContainer, Container } from './styles'
+
+
+type Props = {
+  orders: Order[]
+}
+
+const MHProductsList: React.FC<Props> = ({ orders }: Props) => {
+  const [currentProduct, setCurrentProduct] = useState(0)
```
Comment (aviggiano) [inline components/common/MHProductsList/index.tsx:16]: Change to `currentOrderIndex`, so that it's clearer that it is an `index` about an `order`

**Review by aviggiano — COMMENTED** [review]
Top

File: components/common/FilterMarketplace/Filters.tsx (hunk 1 above)
Comment (pedrohw12) [reply components/common/FilterMarketplace/Filters.tsx]: For sure. Created just now

File: components/common/FilterMarketplace/styles.tsx (hunk 2 above)
Comment (pedrohw12) [reply components/common/FilterMarketplace/styles.tsx:10]: The best is to leave the components as much clean as they can be and move css to the styled components.
And in this case, no reason to use css classes if we have the feature of props with styled components

File: components/common/MHProductsList/index.tsx (hunk 3 above)
Comment (pedrohw12) [reply components/common/MHProductsList/index.tsx:32]: The list of Mouse Haunt products. Below the 'Trending Topics'

**Review by juo-dev — APPROVED** [review]

**Pull request merged**